homeassistant/components/media_player/horizon.py

```diff
@@ -0,0 +1,188 @@
+            self._state = STATE_OFF
+
+    def turn_on(self):
+        """Turn the device on."""
+        if self._state is STATE_OFF:
+            self._send_key(self._keys.POWER)
+
+    def turn_off(self):
+        """Turn the device off."""
+        if self._state is not STATE_OFF:
+            self._send_key(self._keys.POWER)
+
+    def media_previous_track(self):
+        """Channel down."""
+        self._send_key(self._keys.CHAN_DOWN)
+        self._state = STATE_PLAYING
+
+    def media_next_track(self):
+        """Channel up."""
+        self._send_key(self._keys.CHAN_UP)
+        self._state = STATE_PLAYING
+
+    def media_play(self):
+        """Send play command."""
+        self._send_key(self._keys.PAUSE)
+        self._state = STATE_PLAYING
+
+    def media_pause(self):
+        """Send pause command."""
+        self._send_key(self._keys.PAUSE)
+        self._state = STATE_PAUSED
+
+    def media_play_pause(self):
+        """Send play/pause command."""
+        self._send_key(self._keys.PAUSE)
+        if self._state == STATE_PAUSED:
+            self._state = STATE_PLAYING
+        else:
+            self._state = STATE_PAUSED
+
+    def play_media(self, media_type, media_id, **kwargs):
+        """Play media / switch to channel."""
+        if MEDIA_TYPE_CHANNEL == media_type:
+            try:
+                self._select_channel(int(media_id))
+                self._state = STATE_PLAYING
+            except ValueError:
+                _LOGGER.error("Invalid channel: %s", media_id)
+        else:
+            _LOGGER.error("Invalid media type %s. Supported type: %s",
+                          media_type, MEDIA_TYPE_CHANNEL)
+
+    def _select_channel(self, channel):
+        """Select a channel (taken from einder library, thx)."""
+        self._send(channel=channel)
+
+    def _send_key(self, key):
+        """Send a key to the Horizon device."""
+        self._send(key=key)
+
+    def _send(self, key=None, channel=None):
+        """Send a key to the Horizon device."""
+        from einder.exceptions import AuthenticationError
+
+        try:
+            if key:
+                self._client.send_key(key)
+            elif channel:
+                self._client.select_channel(channel)
+        except OSError as msg:
+            _LOGGER.error("%s disconnected: %s. Trying to reconnect...",
+                          self._name, msg)
+
+            # for reconnect, first gracefully disconnect
+            self._client.disconnect()
+
+            try:
+                self._client.connect()
+                self._client.authorize()
+            except AuthenticationError as msg:
+                _LOGGER.error("Authentication to %s failed: %s", self._name,
+                              msg)
+                return
+            except OSError as msg:
+                # occurs if horizon box is offline
+                _LOGGER.error("Reconnect to %s failed: %s", self._name, msg)
+                raise PlatformNotReady
```

Comment:
We're not inside `setup_platform` anymore here, so raising `PlatformNotReady` doesn't work. Probably just `return`.